src/main/java/net/authorize/util/XmlUtility.java

```diff
@@ -38,15 +42,23 @@ private XmlUtility() {
 	 * @throws IOException if errors during serialization
 	 * @throws JAXBException if errors during serialization
 	 */
-	public static <T extends Serializable> String getXml(T entity) throws IOException, JAXBException
+	public static synchronized <T extends Serializable> String getXml(T entity) throws IOException, JAXBException
 	{

```

Comment:
The JAXBContext objects access are being synchronized. JAXBContext had been behaving weirdly wiht multi-threading somehow blocking all the calls until last thread is processed and we saw that similar issues have been reported about JAXB earlier as well.